src/serialization/src/main/kotlin/org/vaccineimpact/api/serialization/RuleBasedSerializer.kt

```diff
@@ -38,4 +44,58 @@ fun <T> removeFieldIfNull(original: T, json: JsonObject,
     {
         json.remove(serializer.convertFieldName(property.name))
     }
+}
+
+class BurdenEstimateDataSeriesTypeAdaptor : TypeAdapter<BurdenEstimateDataSeries>()
```

Comment:
Ok, so to avoid iterating over objects multiple times I've written a custom serializer that writes out exactly what we need, based on the "groupBy" property of the parent object. This should be maximally efficient!